sync/sync.go

```diff
@@ -600,6 +600,14 @@ func (s *Synchronizer) PendingData() (core.PendingData, error) {
 
 	p := *ptr
 	if p.Validate(head) {
+		// If pending data is PreConfirmed and it's latest + 1, return it without pre_latest
+		if p.Variant() == core.PreConfirmedBlockVariant && head != nil {
+			preConfirmed := p.(*core.PreConfirmed)
+			if preConfirmed.Block.Number == head.Number+1 && preConfirmed.PreLatest != nil {
```

Comment:
Can you add the meaning of what this `if` condition represents, either as a comment or a variable? 

Is this the case when pending has pre-latest attached, but pre-latest is already latest?